cloud/services/compute/loadbalancers/reconcile.go

```diff
@@ -525,6 +525,9 @@ func (s *Service) createOrGetInternalAddress(ctx context.Context, lbname string)
 		log.Error(err, "Error getting subnet for Internal Load Balancer")
 		return nil, err
 	}
+	if s.scope.ControlPlaneEndpoint().Host != "" {
+		addrSpec.Address = s.scope.ControlPlaneEndpoint().Host
```

Comment:
I have a couple of questions/comments.

1. The Control Plane Endpoint is set using the function here: https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/main/cloud/services/compute/loadbalancers/reconcile.go#L242. It looks like this is called after the function where you are checking for the Control plane endpoint. It probably wont be set yet.
2. The ip address can be set after this call. Are we concerned about getting this address? https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/main/cloud/services/compute/loadbalancers/reconcile.go#L540

Reply:
If you think it would be better to, for example, add a new field `ipAddress` under the internalLoadBalancer spec https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/main/api/v1beta1/types.go#L372 and use that as an option to predefine IP for internal load balancer instead of the current approach, I can do that.

Reply:
@RnkeZ Sorry for the delay, I have been working on some migrations between SDKs and it has been all time consuming. 

I like this approach more. The load balancer info is set as part of the Cluster Spec which to me makes more sense. @RnkeZ and @damdo WDYT?

Reply:
Okay, I'll update the PR with this second approach, where we set the address in the internalLoadBalancer spec.

Reply:
Thanks both, what @barbacbd suggests sounds good to me